### PR TITLE
refactor: migrate to vite from create-react-app

### PR DESCRIPTION
**Pull request by mvadari**
## High Level Overview of Change

This PR migrates the build system from create-react-app and webpack to vite. 

Most of the file changes are switching env variables from `process.env.REACT_APP_...` to `process.env.VITE_...`, and most of the line changes come from `package-lock.json`.

### Context of Change

Vite is a better build system that requires less custom hacky code to get it to work properly. It also builds faster (down to 15-30 seconds from ~1.5 minutes in CI).

### Type of Change

- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Refactor (non-breaking change that only restructures code)

## Before / After

It runs the same.

## Test Plan

Works locally and in CI.

## Future Steps

Test runtime can probably be improved.

[This](https://github.com/facebook/create-react-app/blob/v4.0.3/packages/react-scripts/scripts/test.js) is the script that CRA uses for its tests, so perhaps there is more mocking that we can take from there.

### CODE REVIEW

File: cssTransform.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+module.exports = {
```
Comment (ckniffen) [inline cssTransform.js:1]: Does every transform need to be in a separate file?  Can they be moved to a folder called like `tooling`?  This would declutter the root.

File: src/rippled/lib/utils.js [hunk 2]
```diff
@@ -96,8 +96,6 @@ function RippledError(message, code) {
   this.code = code
 }
 
-require('util').inherits(RippledError, Error)
```
Comment (ckniffen) [inline src/rippled/lib/utils.js:99]: Why was this removed?

File: src/rippled/lib/txSummary/formatAmount.js [hunk 3]
```diff
@@ -1,4 +1,4 @@
-module.exports = (d) => {
+const formatAmount = (d) => {
```
Comment (ckniffen) [inline src/rippled/lib/txSummary/formatAmount.js:1]: could just export on this line.

File: src/rippled/lib/utils.js (hunk 2 above)
Comment (mvadari) [reply src/rippled/lib/utils.js:99]: It was complaining when I ran `npm start`

Comment (mvadari) [reply src/rippled/lib/utils.js:99]: I also don't really understand what it's doing

Comment (ckniffen) [reply src/rippled/lib/utils.js:99]: It adds all properties of Error to RippledError.

Comment (mvadari) [reply src/rippled/lib/utils.js:99]: Wouldn't it be easier to do that with normal inheritance?

Comment (ckniffen) [reply src/rippled/lib/utils.js:99]: That works I just want to make sure the error properties/methods are there.

Comment (mvadari) [reply src/rippled/lib/utils.js:99]: I don't really know how to add this back in - some help would be appreciated.

File: tsconfig.json [hunk 4]
```diff
@@ -10,8 +10,9 @@
     "noUnusedLocals": true,
     "noUnusedParameters": true,
     "noImplicitReturns": true,
+    "noImplicitAny": false,
     "noFallthroughCasesInSwitch": true,
-    "moduleResolution": "node",
+    "moduleResolution": "Node",
```
Comment (ckniffen) [inline tsconfig.json:15]: This should still be lowercase.

**Review by ckniffen — APPROVED** [review]

File: src/setupTests.ts [hunk 5]
```diff
@@ -1,19 +1,25 @@
+import 'dotenv/config'
 import 'jest-enzyme'
 import { configure } from 'enzyme'
 import Adapter from 'enzyme-adapter-react-17-updated'
 
 const mockStorage = {}
 
+// @ts-ignore
```
Comment (shadymoses) [inline src/setupTests.ts:8]: Could also be `ts-expect-error`. Same for all of these.

File: src/index.html [hunk 6]
```diff
@@ -36,6 +28,11 @@
     You need to enable JavaScript to run this app.
   </noscript>
   <div class="xrpl-explorer" id="xrpl-explorer"></div>
+  <script type="module">
+    import { Buffer } from "buffer";
+    window.Buffer = Buffer;
```
Comment (shadymoses) [inline src/index.html:33]: You shouldn't need this with the setup you have in `vite.config`. Was it causing problems without this script?

**Review by shadymoses — APPROVED** [review]
